test/js_test.go

```diff
@@ -2741,6 +2741,100 @@ func TestJetStreamSubscribe_AckPolicy(t *testing.T) {
 
 		checkAcks(t, sub)
 	})
+
+	t.Run("Nak with delay", func(t *testing.T) {
+		js.Publish("bar", []byte("msg"))
+		sub, err := js.SubscribeSync("bar", nats.Durable("nak_dur"))
+		if err != nil {
+			t.Fatalf("Error on subscribe: %v", err)
+		}
+		msg, err := sub.NextMsg(time.Second)
+		if err != nil {
+			t.Fatalf("Error on NextMsg: %v", err)
+		}
+		if err := msg.Nak(nats.NakDelay(500 * time.Millisecond)); err != nil {
```

Comment:
Hmm.. Looking at this suggest maybe NakWithDelay(d time.Duration) might be better? WDYT?

Reply:
Sure. But if we were to add more options to Nak() (which is unlikely), then it may be weird too. Will change for now.